OptionArray.php

```diff
@@ -1,27 +1,36 @@
 <?php
-namespace WP_Rocket\Admin;
+namespace WPMedia\Options;
 
 /**
- * Manages the data inside an option.
+ * Manages the array data coming from an option.
  *
  * @since 3.0
  * @author Remy Perona
  */
-class Options_Data {
+class OptionArray {
```

Comment:
@Tabrisrp Check out implementing `ArrayAccess`, which allows the objects to access it as an array. It then fits with the new name `OptionArray`.

```php
use AccessArray;

class OptionArray implements AccessArray {
```

[See the docs here](https://www.php.net/manual/en/class.arrayaccess.php).

You'll need to add 4 methods to allow array iteration:

- `offsetSet`
- `offsetExists`
- `offsetUnset`
- `offsetGet`

Laravel is liberal in its use of `ArrayAccess`. Check out how they do it with things like the [`Container`](https://github.com/laravel/framework/blob/0a70bebd5ecfd51185a312bbfb60ee7f8ff7eb09/src/Illuminate/Container/Container.php#L15), [`Config`](https://github.com/laravel/framework/blob/0a70bebd5ecfd51185a312bbfb60ee7f8ff7eb09/src/Illuminate/Config/Repository.php#L9) and [`Cache\Repository`](https://github.com/laravel/framework/blob/0a70bebd5ecfd51185a312bbfb60ee7f8ff7eb09/src/Illuminate/Cache/Repository.php).

Reply:
Would you implement it in addition to the methods already existing in the class, or in replacement of them?

Reply:
In addition to.... These new ones call the existing methods.